test/e2e/autoscale_test.go

```diff
@@ -128,50 +128,43 @@ func generateTraffic(ctx *testContext, concurrency int, duration time.Duration,
 	return nil
 }
 
-func generateTrafficAtFixedRPS(ctx *testContext, rps float64, duration time.Duration, stopChan chan struct{}) error {
+func generateTrafficAtFixedRPS(ctx *testContext, rps int, duration time.Duration, stopChan chan struct{}) error {
 	var (
 		totalRequests      int32
 		successfulRequests int32
 	)
 
-	ctx.t.Logf("Maintaining %v RPS requests for %v.", rps, duration)
-	client, err := pkgTest.NewSpoofingClient(ctx.clients.KubeClient, ctx.t.Logf, ctx.domain, test.ServingFlags.ResolvableDomain)
-	if err != nil {
-		return fmt.Errorf("error creating spoofing client: %v", err)
+	rate := vegeta.Rate{Freq: rps, Per: time.Second}
+	target := vegeta.Target{
+		Method: "GET",
+		URL:    fmt.Sprintf("http://%s?sleep=100", ctx.domain),
 	}
+	targeter := vegeta.NewStaticTargeter(target)
+	attacker := vegeta.NewAttacker(vegeta.Timeout(duration))
 
-	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("http://%s?sleep=100", ctx.domain), nil)
-	if err != nil {
-		return fmt.Errorf("error creating HTTP request: %v", err)
-	}
+	ctx.t.Logf("Maintaining %v RPS requests for %v.", rps, duration)
+
+	// Start the attack!
+	results := attacker.Attack(targeter, rate, duration, "load-test")
 
-	ticker := time.NewTicker(time.Second / time.Duration(rps))
-	defer ticker.Stop()
-	done := time.After(duration)
 	for {
 		select {
 		case <-stopChan:
 			ctx.t.Log("Stopping generateTraffic")
 			return nil
-		case <-done:
-			ctx.t.Log("Time is up; done")
-			return nil
-		case <-ticker.C:
-			go func() {
-				atomic.AddInt32(&totalRequests, 1)
-				res, err := client.Do(req)
-				if err != nil {
-					ctx.t.Logf("Error making request %v", err)
-					return
-				}
+		case res, ok := <-results:
+			if !ok {
+				ctx.t.Log("Time is up; done")
+				return nil
+			}
 
-				if res.StatusCode != http.StatusOK {
-					ctx.t.Logf("Status = %d, want: %d", res.StatusCode, http.StatusOK)
-					ctx.t.Logf("Response: %s", res)
-					return
-				}
-				atomic.AddInt32(&successfulRequests, 1)
-			}()
+			atomic.AddInt32(&totalRequests, 1)
+			if res.Code != http.StatusOK {
+				ctx.t.Logf("Status = %d, want: %d", res.Code, http.StatusOK)
```

Comment:
well, there's no need to format status OK, we know what we want :)

Reply:
Done.